lib/runner/cli/cli.js

```diff
@@ -204,6 +211,33 @@ class CliRunner {
     return this;
   }
 
+  async launchMobileEmulator() {
+    const {desiredCapabilities} = this.test_settings;
+    const chromeOptions = desiredCapabilities['goog:chromeOptions'];
+
+    if (chromeOptions && chromeOptions.androidPackage) {
+      const {getPlatformName, getAlreadyRunningAvd, launchAVD} = require('nightwatch-mobile-helper');
+      const sdkRoot = getSdkRootFromEnv();
+      const platform = getPlatformName();
+      const NIGHTWATCH_AVD = desiredCapabilities.avd || 'nightwatch-android-11';
+  
+      return getAlreadyRunningAvd(sdkRoot, platform, NIGHTWATCH_AVD).then(async (emulatorAlreadyRunning) => {
+        if (!emulatorAlreadyRunning) {
+          await launchAVD(sdkRoot, platform, NIGHTWATCH_AVD);
+        }
+      }).catch(e => {
+        const err = new Error('adb-server not running');
```

Comment:
Failure in running AVD is not necessarily due to adb-server not running. We can make this error more generic, like "Failed to launch AVD inside Android Emulator".